eiger_io/fs_handler.py

```diff
@@ -20,3 +21,45 @@ def __call__(self, seq_id):
         master_path = '{}_{}_master.h5'.format(self._base_path, seq_id)
         # TODO Return a multi-dimensional PIMS seq.
         return np.array(EigerImages(master_path))
+
+
+EIGER_MD_DICT = {
+    'y_pixel_size': 'entry/instrument/detector/y_pixel_size',
+    'x_pixel_size': 'entry/instrument/detector/x_pixel_size',
+    'detector_distance': 'entry/instrument/detector/detector_distance',
+    'incident_wavelength': 'entry/instrument/beam/incident_wavelength',
+    'frame_time': 'entry/instrument/detector/frame_time',
+    'beam_center_x': 'entry/instrument/detector/beam_center_x',
+    'beam_center_y': 'entry/instrument/detector/beam_center_y',
+    'count_time': 'entry/instrument/detector/count_time',
+    'pixel_mask': 'entry/instrument/detector/detectorSpecific/pixel_mask',
+}
+
+
+class LazyEigerHandler(HandlerBase):
+    specs = {'AD_EIGER'} | HandlerBase.specs
+    def __init__(self, fpath, frame_per_point, mapping=None):
+        # create pims handler
+        self.vals_dict = EIGER_MD_DICT.copy()
+        if mapping is not None:
+            self.vals_dict.update(mapping)
+        self._base_path = fpath
+        self.fpp = frame_per_point
+
+    def __call__(self, seq_id):
+        master_path = '{}_{}_master.h5'.format(self._base_path, seq_id)
+        md = {}
+        print('hdf5 path = %s' % master_path)
+        with h5py.File(master_path, 'r') as f:
+            md = {k: f[v][()] for k, v in self.vals_dict.items()}
+        # the pixel mask from the eiger contains:
+        # 1  -- gap
+        # 2  -- dead
+        # 4  -- under-responsive
+        # 8  -- over-responsive
+        # 16 -- noisy
+        pixel_mask = ~(md['pixel_mask'] == 0)
+        md['binary_mask'] = pixel_mask.astype(np.uint16)
```

Comment:
After some (light) experimenting, it would appear that I was wrong and this does not need to be cast to an int at all.

```python
import numpy as np
arr = np.random.random(10)
arr2 = np.random.random(10)
arr = arr > 0.5
print(arr)
print(arr2 * arr)
```
produces
```
[ True  True  True False  True  True  True False False False]
[ 0.50588247  0.53902746  0.72079631  0.          0.19657096  0.77427032
  0.26992042  0.          0.          0.        ]
```

Perhaps just change lines 61 and 62 to be this single line statement -->  `md['binary_mask'] = ~(md['pixel_mask'] == 0)`